src/yaku.js

```diff
@@ -317,14 +358,97 @@ function getShousangen(hand) {
 
 //Daisangen
 function getDaisangen(hand) {
-	if (hand.filter(tile => tile.type == 3 && tile.index == 5).length >= 3 &&
-		hand.filter(tile => tile.type == 3 && tile.index == 6).length >= 3 &&
-		hand.filter(tile => tile.type == 3 && tile.index == 7).length >= 3) {
-		return { open: 10, closed: 10 }; //Yakuman -> 10?
+	if (hand.filter(tile => tile.type === 3 && tile.index === 5).length >= 3 &&
+		hand.filter(tile => tile.type === 3 && tile.index === 6).length >= 3 &&
+		hand.filter(tile => tile.type === 3 && tile.index === 7).length >= 3) {
+		return { open: YAKUMAN_SCORE, closed: YAKUMAN_SCORE };
+	}
+	return { open: 0, closed: 0 };
+}
+
+//Suuankou
+function getSuuankou(triplets) {
+	if (!isConsideringCall) {
+		if (parseInt(triplets.length / 3) >= 4) {
+			return { open: 0, closed: YAKUMAN_SCORE };
+		}
+	}
+	return { open: 0, closed: 0 };
+}
+
+//Tsuuiisou
+function getTsuuiisou(hand, triplets) {
+	if (hand.filter(tile => tile.type === 3).length >= 13) {
+		if (parseInt(triplets.length / 3) >= 3) {
+			return { open: YAKUMAN_SCORE, closed: YAKUMAN_SCORE };
+		}
+	}
+	return { open: 0, closed: 0 };
+}
+
+//Ryuuiisou
+function getRyuuiisou(hand) {
+	if (hand.filter(tile => 
+		(tile.type == 2 && (tile.index === 2
+						|| tile.index === 3
+						|| tile.index === 4
+						|| tile.index === 6
+						|| tile.index === 8))
+			|| (tile.type === 3 && tile.index === 6)).length === hand.length) {
+		return { open: YAKUMAN_SCORE, closed: YAKUMAN_SCORE };
 	}
 	return { open: 0, closed: 0 };
 }
 
+//Chinroutou
+function getChinroutou(hand) {
+	if (hand.find(tile => tile.type === 3 || (tile.index != 1 && tile.index != 9))) {
+		return { open: 0, closed: 0 };
+	} else {
+		return { open: YAKUMAN_SCORE, closed: YAKUMAN_SCORE };
+	}
+}
+
+//Suushiihou
+function getSuushiihou(hand) {
+	if (hand.filter(tile => tile.type === 3 && tile.index <= 4).length == 11) {
+		return { open: YAKUMAN_SCORE, closed: YAKUMAN_SCORE };
+	}
+	return { open: 0, closed: 0 };
+}
+
+//ChuurenPoutou
+function getChuurenPoutou(hand, triplets, sequences) {
```

Comment:
In general it would be nice to have some testcases for these yaku, but I can add these myself later.

Reply:
Remove it.